File: lib_before_phosphor_fix/presentation/tracker_screen/tracker/android_native/java/MainActivity_bonded_ble_bridge_snippet.java
// Java-вариант для android/app/src/main/java/.../MainActivity.java.
// Package должен остаться тот же, что уже есть в вашем MainActivity.java.

import android.Manifest;
import android.bluetooth.BluetoothAdapter;
import android.bluetooth.BluetoothDevice;
import android.bluetooth.BluetoothManager;
import android.content.pm.PackageManager;
import android.os.Build;

import androidx.annotation.NonNull;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import io.flutter.embedding.android.FlutterActivity;
import io.flutter.embedding.engine.FlutterEngine;
import io.flutter.plugin.common.MethodChannel;

public class MainActivity extends FlutterActivity {
    private static final String ACTION_TRACKER_BLE_CHANNEL = "sportoteka/action_tracker_ble";

    @Override
    public void configureFlutterEngine(@NonNull FlutterEngine flutterEngine) {
        super.configureFlutterEngine(flutterEngine);

        new MethodChannel(flutterEngine.getDartExecutor().getBinaryMessenger(), ACTION_TRACKER_BLE_CHANNEL)
                .setMethodCallHandler((call, result) -> {
                    if ("getBondedBluetoothDevices".equals(call.method)) {
                        try {
                            result.success(getBondedBluetoothDevicesForTracker());
                        } catch (SecurityException e) {
                            result.error("bluetooth_permission", e.getMessage(), null);
                        } catch (Exception e) {
                            result.error("bluetooth_bonded_error", e.getMessage(), null);
                        }
                    } else {
                        result.notImplemented();
                    }
                });
    }

    private List<Map<String, Object>> getBondedBluetoothDevicesForTracker() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S
                && checkSelfPermission(Manifest.permission.BLUETOOTH_CONNECT) != PackageManager.PERMISSION_GRANTED) {
            throw new SecurityException("Нет разрешения BLUETOOTH_CONNECT / Устройства поблизости");
        }

        BluetoothManager manager = (BluetoothManager) getSystemService(BluetoothManager.class);
        BluetoothAdapter adapter = manager != null ? manager.getAdapter() : BluetoothAdapter.getDefaultAdapter();
        List<Map<String, Object>> result = new ArrayList<>();
        if (adapter == null || adapter.getBondedDevices() == null) return result;

        for (BluetoothDevice device : adapter.getBondedDevices()) {
            Map<String, Object> item = new HashMap<>();
            item.put("address", device.getAddress());
            item.put("id", device.getAddress());
            item.put("name", device.getName() == null ? "" : device.getName());
            item.put("type", bluetoothTypeName(device.getType()));
            item.put("bondState", bondStateName(device.getBondState()));
            result.add(item);
        }
        return result;
    }

    private String bluetoothTypeName(int type) {
        switch (type) {
            case BluetoothDevice.DEVICE_TYPE_CLASSIC: return "classic";
            case BluetoothDevice.DEVICE_TYPE_LE: return "le";
            case BluetoothDevice.DEVICE_TYPE_DUAL: return "dual";
            default: return "unknown";
        }
    }

    private String bondStateName(int state) {
        switch (state) {
            case BluetoothDevice.BOND_BONDED: return "bonded";
            case BluetoothDevice.BOND_BONDING: return "bonding";
            case BluetoothDevice.BOND_NONE: return "none";
            default: return String.valueOf(state);
        }
    }
}

File: lib_before_phosphor_fix/presentation/tracker_screen/tracker/android_native/kotlin/MainActivity_bonded_ble_bridge_snippet.kt
// Вставьте этот блок в ваш android/app/src/main/kotlin/.../MainActivity.kt.
// Package должен остаться тот же, что уже есть в вашем MainActivity.kt.
// Если MainActivity уже переопределяет configureFlutterEngine — просто добавьте
// MethodChannel(...) внутрь существующего configureFlutterEngine после super.

import android.Manifest
import android.bluetooth.BluetoothAdapter
import android.bluetooth.BluetoothDevice
import android.bluetooth.BluetoothManager
import android.content.pm.PackageManager
import android.os.Build
import io.flutter.embedding.android.FlutterActivity
import io.flutter.embedding.engine.FlutterEngine
import io.flutter.plugin.common.MethodChannel

class MainActivity : FlutterActivity() {
    private val actionTrackerBleChannel = "sportoteka/action_tracker_ble"

    override fun configureFlutterEngine(flutterEngine: FlutterEngine) {
        super.configureFlutterEngine(flutterEngine)

        MethodChannel(flutterEngine.dartExecutor.binaryMessenger, actionTrackerBleChannel)
            .setMethodCallHandler { call, result ->
                when (call.method) {
                    "getBondedBluetoothDevices" -> {
                        try {
                            result.success(getBondedBluetoothDevicesForTracker())
                        } catch (e: SecurityException) {
                            result.error("bluetooth_permission", e.message ?: "Нет Bluetooth-разрешения", null)
                        } catch (e: Exception) {
                            result.error("bluetooth_bonded_error", e.message ?: e.toString(), null)
                        }
                    }
                    else -> result.notImplemented()
                }
            }
    }

    private fun getBondedBluetoothDevicesForTracker(): List<Map<String, Any?>> {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S &&
            checkSelfPermission(Manifest.permission.BLUETOOTH_CONNECT) != PackageManager.PERMISSION_GRANTED
        ) {
            throw SecurityException("Нет разрешения BLUETOOTH_CONNECT / Устройства поблизости")
        }

        val manager = getSystemService(BluetoothManager::class.java)
        val adapter = manager?.adapter ?: BluetoothAdapter.getDefaultAdapter() ?: return emptyList()
        val bonded = adapter.bondedDevices ?: return emptyList()

        return bonded.map { device ->
            mapOf(
                "address" to device.address,
                "id" to device.address,
                "name" to (device.name ?: ""),
                "type" to bluetoothTypeName(device.type),
                "bondState" to bondStateName(device.bondState),
                "uuids" to (device.uuids?.map { it.uuid.toString() } ?: emptyList<String>())
            )
        }
    }

    private fun bluetoothTypeName(type: Int): String {
        return when (type) {
            BluetoothDevice.DEVICE_TYPE_CLASSIC -> "classic"
            BluetoothDevice.DEVICE_TYPE_LE -> "le"
            BluetoothDevice.DEVICE_TYPE_DUAL -> "dual"
            else -> "unknown"
        }
    }

    private fun bondStateName(state: Int): String {
        return when (state) {
            BluetoothDevice.BOND_BONDED -> "bonded"
            BluetoothDevice.BOND_BONDING -> "bonding"
            BluetoothDevice.BOND_NONE -> "none"
            else -> state.toString()
        }
    }
}
